app.js

```diff
@@ -97,12 +111,34 @@ const fmt = {
 	}
 }
 
-const render = () => {
+const list_render = () => {
 	const TEMPLATE = document.getElementById('template-event-item').innerHTML
 	document.getElementById('event-list').innerHTML = ''
 
-	for (var i = 0; i < DATA.length; i++) {
-		let event = DATA[i]
+	let visible_events = DATA
+
+	// School filter
+	if (FILTER.school) {
+		visible_events = visible_events.filter((event) => {
+			return (event.contestants.min.substr(0, 2) == FILTER.school || event.contestants.max.substr(0, 2) == FILTER.school)
+		})
+	}
+
+	// Sciences filter
+	if (FILTER.sciences.length != 0) {
+		visible_events = visible_events.filter((event) => {
+			for (var i = FILTER.sciences.length - 1; i >= 0; i--) {
+				if (event.sciences.indexOf(FILTER.sciences[i]) === -1) {
+					return false
+				}
+			}
+
+			return true
+		})
+	}
+
+	for (var i = 0; i < visible_events.length; i++) {
+		let event = visible_events[i]
```

Comment:
please let's either use var or let everywhere :D 

https://caniuse.com/let suggests that let should be fine as well.. but you can pick the one you like more

Reply:
oh :slightly_smiling_face: IDE snippet s var-om